src/cosmosdb-preview/azext_cosmosdb_preview/custom.py

```diff
@@ -72,6 +76,168 @@ def _handle_exists_exception(cloud_error):
+                                    location,
+                                    tags=None,
+                                    create_mode=CreateMode.CreateMode.DEFAULT,
+                                    restore_point_in_time_utc=None,
+                                    restore_source_resource_id=None,
+                                    server_version="5.0",
+                                    shard_node_sku=None,
+                                    shard_node_disk_size_gb=None, 
+                                    shard_enable_ha=None,
+                                    shard_node_name=None,
+                                    shard_kind=NodeKind.SHARD,
+                                    shard_node_count=1):
+
+    '''Creates an Azure Cosmos DB Mongo Cluster '''
+
+    if administrator_login is None:
+        raise InvalidArgumentValueError('Initial Mongo Cluster Admin user is required.')
+
+    if administrator_login_password is not None :
+        raise InvalidArgumentValueError('Initial Mongo CLuster Admin Password is required.')
+
+    if restore_point_in_time_utc is not None & restore_source_resource_id is None:
+        raise InvalidArgumentValueError('Both(restore_point_in_time_utc and restore_source_resource_id) Mongo Cluster restore parameters must be provided together')
+
+    if restore_point_in_time_utc is None & restore_source_resource_id is not None:
+        raise InvalidArgumentValueError('Both(restore_point_in_time_utc and restore_source_resource_id) Mongo Cluster restore parameters must be provided together.')
+    
+    mongocluster_restore_parameters = MongoClusterRestoreParameters(
+        point_in_time_utc=restore_point_in_time_utc,
+        source_resource_id=restore_source_resource_id,
+    )
+
+    node_group_spec = NodeGroupSpec(
+        sku= shard_node_sku,
+        disk_size_gb= shard_node_disk_size_gb,
+        enable_ha=shard_enable_ha,
+        name= shard_node_name,
+        kind=shard_kind,
+        node_count= shard_node_count
+    )
+    
+    node_group_specs = list(node_group_spec)
+    mongodb_cluster = MongoCluster(
+        location=location,
+        tags=tags,
+        create_mode=create_mode,
+        restore_parameters= mongocluster_restore_parameters,
+        administrator_login=administrator_login,
+        administrator_login_password=administrator_login_password,
+        server_version=server_version,
+        node_group_specs=node_group_specs)
+
+    return client.begin_create_or_update(resource_group_name, mongocluster_name, mongodb_cluster)
+
+def cli_cosmosdb_mongocluster_update(client,
+                                        resource_group_name,
+                                        mongocluster_name,
+                                        administrator_login,
+                                        administrator_login_password,
+                                        tags=None,
+                                        create_mode=CreateMode.CreateMode.DEFAULT,
```

Comment:
```suggestion
                                        create_mode=CreateMode.DEFAULT.value,
```

Reply:
Besides changing the default value, you need to register this enum arguemnt in `param.py`:
```
c.argument('create_mde', argument_type=get_enum_type(CreateMode))
```

Reply:
fixed